contracts/Rentals.sol

```diff
@@ -425,7 +425,7 @@ contract Rentals is
             // Check that the caller is the contract defined in the offer to ensure the function is being
             // called through an ERC721.safeTransferFrom.
             // Also check that the token id is the same as the one provided in the offer.
-            require(msg.sender == offer.contractAddress && offer.tokenId == _tokenId, "Rentals#onERC721Received: ASSET_MISMATCH");
+            require(_msgSender() == offer.contractAddress && offer.tokenId == _tokenId, "Rentals#onERC721Received: ASSET_MISMATCH");
```

Comment:
Hmmmm I would prefer keeping this as it is. It is a security definition to only use msg.sender in this method.